Add schema lifecycle and versioning policy documents (#32)
SeequentEvo/evo-schemas: docs/versioning-and-release-process/schema-development-lifecycle.md

The process has been designed to make it as easy as possible for submitters who come with capacity to help do the work—including, but not limited to the case where the submitter is a Seequent Product Manager and the work is done by Seequent employees. In this case, the Schema Working Group will work along with the developers to guide the schema development process and fit it into the overall set of schemas. This might include work like recommending where to break shared capabilities out into a Component.

In the event there is a request for a new schema but no capacity to build it, the RFC might not be accepted into the **Solution** phase at all; or, work might pause or halt entirely. This doesn't mean it's a bad idea, nor does it mean it won't be worked on at all—however, capacity is limited. Wherever possible, we will communicate clearly about these decisions.

### Publication

> The **Publication** phase tracks the schema as it makes its way to being publicly available. In the event that a schema has been developed in private, it must be disclosed commercially for it to enter the **Tech Preview** state. Functionally, a schema that is in **Tech Preview** is fully usable within the running instances of the [Geoscience Objects API](https://developer.seequent.com/docs/api/geoscience-object/geoscience-object-api). This enables development against the schema by any Evo user (including Seequent developers, customers and partners).

Reviewer: ```suggestion
The **Publication** phase tracks the schema as it makes its way to being publicly available. In the event that a schema has been developed in private, it must be disclosed commercially for it to enter the **Tech Preview** state. Functionally, a schema that is in **Tech Preview** is fully usable within the running instances of the Geoscience Object API. This enables development against the schema by any Evo user (including Seequent developers, customers and partners).
```

> The **Publication** phase tracks the schema as it makes its way to being publicly available. In the event that a schema has been developed in private, it must be disclosed commercially for it to enter the **Tech Preview** state. Functionally, a schema that is in **Tech Preview** is fully usable within the running instances of the [Geoscience Objects API](https://developer.seequent.com/docs/api/geoscience-object/geoscience-object-api). This enables development against the schema by any Evo user (including Seequent developers, customers and partners).

Reviewer: ```suggestion
The **Publication** phase tracks the schema as it makes its way to being publicly available. In the event that a schema has been developed in private, it must be disclosed commercially for it to enter the **Tech Preview** state. Functionally, a schema that is in **Tech Preview** is fully usable within the running instances of the [Geoscience Object API](https://developer.seequent.com/docs/api/geoscience-object/geoscience-object-api). This enables development against the schema by any Evo user (including Seequent developers, customers and partners).
```

For schemas in **Tech Preview**, no Service Level Agreements (SLA) or Deprecation guarantees apply.

Customers and partners should expect that Seequent products have mixed support for schemas in this state—part of the purpose is to give teams a chance to build against the schema, and this includes Seequent / Bentley internal teams. For example, common features like visualization might not work yet for schemas in **Tech Preview**, even if it seems like an obvious feature. Mature products might not be able to read or write the schema.

**Tech Preview** schemas are marked as such in the Developer Portal, and that is the authoritative place to check the status of a schema you wish to use.

### Operation

For schemas that are fully **Supported**, the usual Evo SLAs and Deprecation policy apply. That doesn't mean they are supported everywhere (which schemas are supported depends on the product and workflow).

Sometimes, a **Supported** schema might no longer be useful (perhaps it has been superseded or didn't receive sufficient uptake in products). In that situation, the schema will enter the **Deprecated** state, which means we recommend that it not be used for new integrations.

> In the event that a schema is **Deprecated**, Seequent will communicate any proposed changes to the support, including a clear estimate of when and how anything might change.

Reviewer: Is `to the support` intended in this sentence?

Author: Yeah, this is unclear – let me rephrase
```suggestion
In the event that a schema or version of a schema is **Deprecated**, Seequent recommends against using the deprecated schema for new integrations. Seequent will communicate any planned changes to the availability of the schema within the Geoscience Object API, along with their timeline. If the deprecation of a schema impacts your workflow, please reach out to the Seequent Evo support team or Developer Community Champions.
```
